### ISSUE TITLE
Godot 4, Filter Scripts layout is confusing when switching between scenes, GUI layout improvements?

### ISSUE THREAD

**Issue by BdR76** [godotengine/godot-proposals#7838]
### Describe the project you are working on

I'm new to Godot and it's a very cool game development tool, but the way the code editor IDE is setup, specifically the **Filter Scripts** panel is confusing.

When switching between scenes, I've often found myself editing the incorrect script file. So for example, when I want to add a new function to the player scene script, I'll accidentally add it to the script of the main scene.

![godot_editor_tabs_1](https://github.com/godotengine/godot-proposals/assets/7793956/4a20a1e8-ff82-4b0a-9d4e-78ad72a571e9)

### Describe the problem or limitation you are having in your project

What's confusing to me, is that when the code editor is open and you switch between different scenes, it does not necessarily correspond with a script associated with that scene. You still have to select the script file independently from the scene tab. I understand that a scene can have several scripts associated with it, but as far as I can see the filter script area will just show all scripts in the project, not the specific scripts for that scene.

See the example screen shot above, I've selected the player character scene (my_guy) but it shows the script from the gamescene.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I don't have a complete proposal for a solution to this because it would involve redesigning part of the editor, but maybe it's possible to make the filter only show the relevant scripts for the scene? Or alternatively, is it possible to visually setup the IDE in a different way?

![godot_editor_tabs_2](https://github.com/godotengine/godot-proposals/assets/7793956/f258867e-2035-467a-83bb-621782329f68)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Like I said I don't know an exact solution because this requires a re-design of the Godot code editor. Btw I've looked but couldn't find any options to remove the filter area. btw I'm working with Godot 4.1.1

### If this enhancement will not be used often, can it be worked around with a few lines of script?

A possible workaround is to work separate editor like VS Code, but that defeats the purpose and advantages of the Godot all-in-one IDE

### Is there a reason why this should be core and not an add-on in the asset library?

Because it's part of the core Godot engine and editor

**Comment by Zireael07** [comment]
The tabs at the top are not related to the filters nor scripts. They're your open scenes. You can have a different scene and a different script open.

**Comment by BdR76** [comment]
Yes, I understand that that is the way the GUI layout has always been so far, but I'm saying that it's something that will easily confuse newcomers and cause mistakes, because it's not intuitive.

I was trying to discuss potential different ways you could switch between nodes and scripts. For example it might be improved if the Filter Scripts only shows the scripts relevent to that node, or maybe if the script editor was its own separate pinned tab or something like that. Btw I also posted [this question here](https://www.reddit.com/r/godot/comments/16l1qg0/godot_4_code_editor_and_filter_scripts_can_be/) and at least some people seem to agree

**Comment by KoBeWi** [comment]
Related #7098
Note that the script switching when changing scenes can be disabled in editor settings (`open_dominant_script_on_scene_change`).

btw the "filter area" is just a script list. Filter Scripts is an input field for filtering scripts by name, it's not a header (although it does look like one a bit 🤔)

**Comment by YeldhamDev** [comment]
I (and likely most devs too) often jump around multiple scripts in my projects, having them being hidden until I open/switch a scene would be a big detriment to overall productivity.

Maybe a better solution would be to highlight the scripts of the current scene in a different way?

**Comment by BdR76** [comment]
> Related #7098

imho that post is a pretty accurate description of the same issue I'm trying to describe, and maybe it's telling that even long-time users experience this too, not just newcomers.

> [This was](https://github.com/godotengine/godot-proposals/issues/7098#issuecomment-1594549861) already proposed in #966, #4457, and #5314. And to a degree in #4081.

Also, if this has been brought up multiple times, it's probably not something that can be solved with a quick-fix here or a small change there. Frankly, it's up to the designer/product-owner to decide what could be done about this.

**Comment by thygrrr** [comment]
Check out the "Script IDE" extension in the Asset Library.

It already improves this a lot by making everything a tab. And using the old scripts pane as a semantic outline of the CURRENT script.
